src/js/utils.js

```diff
@@ -55,74 +57,144 @@ export function parseShorthand(obj, props) {
+ * Determines options for the Tippy.js tooltip and initializes it.
  */
-export function setupPopper() {
-  if (isUndefined(Popper)) {
-    throw new Error('Using the attachment feature of Shepherd requires the Popper.js library');
+export function setupTooltipElem() {
+  if (isUndefined(tippy)) {
+    throw new Error(missingTippy);
+  }
+
+  if (this.tooltipElem) {
+    this.tooltipElem.destroy();
   }
 
-  const opts = this.getAttachTo();
-  opts.modifiers = opts.modifiers || {};
-  let attachment = opts.on || 'right';
-  opts.positionFixed = false;
+  const attachToOpts = this.parseAttachTo();
+
+  this.tooltipElem = _makeTippyInstance.call(this, attachToOpts);
+
+  this.target = attachToOpts.element || document.body;
+  this.target.classList.add('shepherd-enabled', 'shepherd-target');
+
+  this.el.classList.add('shepherd-element');
+}
 
-  if (isUndefined(opts.element)) {
-    attachment = 'top';
-    _setupCenteredPopper(opts);
+/**
+ * Passes `options.attachTo` to `_parseAttachToOpts` to get the correct `attachTo` format
+ * @returns {({} & {element, on}) | ({})}
+ * `element` is a qualified HTML Element
+ * `on` is a string position value
+ */
+export function parseAttachTo() {
+  const options = _parseAttachToOpts(this.options.attachTo) || {};
+  const returnOpts = Object.assign({}, options);
+
+  if (isString(options.element)) {
+    // Can't override the element in user opts reference because we can't
+    // guarantee that the element will exist in the future.
+    try {
+      returnOpts.element = document.querySelector(options.element);
+    } catch(e) {
+      // TODO
+    }
+    if (!returnOpts.element) {
+      console.error(`The element for this Shepherd step was not found ${options.element}`);
+    }
   }
 
-  if (this.popper) {
-    this.popper.destroy();
+  return returnOpts;
+}
+
+/**
+ * Generates a `Tippy` instance from a set of base `attachTo` options
+ *
+ * @return {tippy} The final tippy instance
+ * @private
+ */
+function _makeTippyInstance(baseAttachToOptions) {
+  if (!baseAttachToOptions.element) {
+    return _makeCenteredTippy.call(this);
   }
 
-  this.el.classList.add('shepherd-element');
-  const popperOpts = _mergePopperOptions.call(this, attachment, opts);
-  this.popper = new Popper(opts.element, this.el, popperOpts);
+  const tippyOptions = _makeAttachedTippyOptions.call(this, baseAttachToOptions);
 
-  this.target = opts.element;
-  this.target.classList.add('shepherd-enabled', 'shepherd-target');
+  return tippy.one(baseAttachToOptions.element, tippyOptions);
 }
 
 /**
- * Merge the global popperOptions, and the local opts
- * @param {String} attachment The direction for attachment
- * @param {Object} opts The local options
- * @return {Object} The merged popperOpts object
+ * Generates the hash of options that will be passed to `Tippy` instances
+ * target an element in the DOM.
+ *
+ * @param {Object} baseAttachToOptions The local `attachTo` options
```

Comment:
Yeah, I think I originally did this because I had part of the code modifying the hash. It doesn't look like I finished that way, though 😄 -- so good call.

Reply:
Updated